lib/metadata/index.js

```diff
@@ -262,6 +281,62 @@ Metadata.prototype.getReplicas = function (keyspaceName, tokenBuffer) {
   return keyspace.replicas[closestToken];
 };
 
+/**
+ * Gets the token ranges that define data distribution in the ring.
+ *
+ * @returns {Set<TokenRange>} The ranges of the ring.
+ */
+Metadata.prototype.getTokenRanges = function () {
+  return this.tokenRanges;
+};
+
+/**
+ * Gets the token ranges that are replicated on the given host, for
+ * the given keyspace.
+ *
+ * @param {String} keyspaceName The name of the keyspace to get ranges for.
+ * @param {Host} host The host.
+ * @returns {Set<TokenRange>} Ranges for the keyspace on this host
+ */
+Metadata.prototype.getTokenRangesForHost = function (keyspaceName, host) {
+  if (!this.ring) {
+    return null;
+  }
+  const self = this;
+  let keyspace;
+  if (keyspaceName) {
+    keyspace = this.keyspaces[keyspaceName];
+    if (!keyspace) {
+      // the keyspace was not found, the metadata should be loaded beforehand
+      return null;
+    }
+  }
+
+  // TODO: This code is duplicated from getReplicas, consider making common method.
+  if (!keyspace.replicas) {
+    //Calculate replicas the first time for the keyspace
+    keyspace.replicas =
+      keyspace.tokenToReplica(this.tokenizer, this.ringTokensAsStrings, this.primaryReplicas, this.datacenters);
+  }
+
+  // TODO: Consider caching/precomputing keyspace -> (host, range) as this could be an expensive operation.
```

Comment:
wdyt?  Should I store something like `keyspace.hostToRanges` similar to how we do something for `keyspace.replicas`?

Reply:
I don't think it would be likely for a user to call `getTokenRangesForHost()` with the same parameters more than once.